### PR TITLE
dellctl- support cluster registration with uids

### PR DESCRIPTION
**Pull request by bpjain2004**
# Description
Now dellctl supports cluster registration with optional uids. This PR is to reflect the same changes in documentation.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| dell/csm#350|

# Checklist:

- [x] Have you run a grammar and spell checks against your submission?
- [x] Have you tested the changes locally?
- [x] Have you tested whether the hyperlinks are working properly?
- [x] Did you add the examples wherever applicable?
- [ ] Have you added high-resolution images?

### CODE REVIEW

**Review by rsedlock1958 — COMMENTED** [review]
I have reviewed this. No edits are required.

File: content/docs/cli/_index.md [hunk 1]
```diff
@@ -165,9 +176,9 @@ List all clusters currently being managed by dellctl
 
 ```
 # dellctl cluster get
-CLUSTER ID      VERSION URL
-cluster1        v1.22   https://1.2.3.4:6443
-cluster2        v1.22   https://1.2.3.5:6443
+CLUSTER ID      VERSION URL                             UID
+cluster1        v1.22   https://10.225.7.48:6443
```
Comment (tdawe) [inline content/docs/cli/_index.md:180]: Mask out internal IPs

Comment (bpjain2004) [reply content/docs/cli/_index.md]: done.

**Review by tdawe — APPROVED** [review]

File: content/docs/cli/_index.md [hunk 2]
```diff
@@ -93,10 +93,12 @@ Add one or more k8s clusters to be managed by dellctl
 ##### Flags
 
 ```
+Flags:
+  -n, --names strings   cluster names
   -f, --files strings   paths for kube config files
+  -u, --uids strings    uids of the kube-system namespaces in the clusters
```
Comment (sharmilarama) [inline content/docs/cli/_index.md:99]: Can we indicate all the flags that are optional ?

Comment (bpjain2004) [reply content/docs/cli/_index.md:99]: We use cobra module for cli development. The same is used by kubectl, velero etc.
The help message is auto generated by the cobra module. Currently it doesn't mark any of the flags as required/optional, in the help message, even though the flag is marked as required in the code. But it errors out if the mandatory flag is not specified on the cli.

For example, the flag "names" is marked as required in the code:
```
	_ = clusterAddCmd.Flags().StringSliceP("names", "n", []string{}, "cluster names")
	_ = clusterAddCmd.MarkFlagRequired("names")

```
When the command is run without this flag:

```
 -> dellctl cluster add
Error: required flag(s) "names", "files" not set
Usage:
  dellctl cluster add [flags]

Examples:
  dellctl cluster add -n <cluster-name> -f <config-file>
  dellctl cluster add -n <cluster-name1> -f <config-file1> -n <cluster-name2> -f <config-file2> ...
  dellctl cluster add -n <cluster-name1,cluster-name2,...> -f <config-file1,config-file1,...>
  dellctl cluster add -n <cluster-name> -f <config-file> -u <cluster-uid>


Flags:
  -n, --names strings   cluster names
  -f, --files strings   paths for kube config files
  -u, --uids strings    uids of the kube-system namespaces in the clusters
      --force           forcefully add cluster
  -h, --help            help for add

required flag(s) "names", "files" not set

```

To overcome this issue we can have custom help messages for each of the commands. We can explore and consider this during the next release.

Comment (sharmilarama) [reply content/docs/cli/_index.md:99]: Ok sounds good. Thanks.

**Review by sharmilarama — APPROVED** [review]

**Review by shanmydell — APPROVED** [review]

**Pull request merged**